src/plugins/mocap/mocap.cpp

```diff
@@ -0,0 +1,230 @@
+{
+    return _impl->set_vision_position_estimate(vision_position_estimate);
+}
+
+Mocap::Result Mocap::set_attitude_position_mocap(AttitudePositionMocap attitude_position_mocap)
+{
+    return _impl->set_attitude_position_mocap(attitude_position_mocap);
+}
+
+Mocap::Result Mocap::set_odometry(const Odometry& odometry)
+{
+    return _impl->set_odometry(odometry);
+}
+
+const char* Mocap::result_str(Result result)
+{
+    switch (result) {
+        case Result::SUCCESS:
+            return "Success";
+        case Result::NO_SYSTEM:
+            return "No system";
+        case Result::CONNECTION_ERROR:
+            return "Connection error";
+        case Result::INVALID_REQUEST_DATA:
+            return "Invalid data in gRPC request";
+        case Result::UNKNOWN:
+        default:
+            return "Unknown";
+    }
+}
+
+bool operator==(const Mocap::Quaternion& lhs, const Mocap::Quaternion& rhs)
+{
+    return lhs.w == rhs.w && lhs.x == rhs.x && lhs.y == rhs.y && lhs.z == rhs.z;
+}
+
+std::ostream& operator<<(std::ostream& str, Mocap::Quaternion const& quaternion)
+{
+    return str << "[w: " << quaternion.w << ", x: " << quaternion.x << ", y: " << quaternion.y
+               << ", z: " << quaternion.z << "]";
+}
+
+bool operator==(const Mocap::VisionPositionEstimate& lhs, const Mocap::VisionPositionEstimate& rhs)
+{
+    // FixMe: Should we check time_usec and reset_counter equality?
+    return (
+        lhs.time_usec == rhs.time_usec && lhs.position_body == rhs.position_body &&
+        lhs.angle_body == rhs.angle_body && lhs.pose_covariance == rhs.pose_covariance);
+}
+
+std::ostream&
+operator<<(std::ostream& str, Mocap::VisionPositionEstimate const& vision_position_estimate)
+{
+    str << "[time_usec: " << vision_position_estimate.time_usec
+        << ", position_body: " << vision_position_estimate.position_body
+        << ", angle_body: " << vision_position_estimate.angle_body
+        << ", pose_covariance: " << vision_position_estimate.pose_covariance
+        << ", reset_counter: " << static_cast<int>(vision_position_estimate.reset_counter) << "]";
+    return str;
+}
+
+bool operator==(const Mocap::AttitudePositionMocap& lhs, const Mocap::AttitudePositionMocap& rhs)
+{
+    // FixMe: Should we check time_usec equality?
+    return (
+        lhs.time_usec == rhs.time_usec && lhs.q == rhs.q &&
+        lhs.position_body == rhs.position_body && lhs.pose_covariance == rhs.pose_covariance);
+}
+
+std::ostream&
+operator<<(std::ostream& str, Mocap::AttitudePositionMocap const& attitude_position_mocap)
+{
+    str << "[time_usec: " << attitude_position_mocap.time_usec
+        << ", q: " << attitude_position_mocap.q
+        << ", position_body: " << attitude_position_mocap.position_body
+        << ", pose_covariance: " << attitude_position_mocap.pose_covariance << "]";
+    return str;
+}
+
+bool operator==(const Mocap::Odometry& lhs, const Mocap::Odometry& rhs)
+{
+    // FixMe: Should we check time_usec, equality?
```

Comment:
I don't know. I guess it's part of the data.

Reply:
Hm. Leave check